tensorboard/webapp/notification_center/_data_source/types.ts

```diff
@@ -0,0 +1,36 @@
+/* Copyright 2021 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+import {Observable} from 'rxjs';
+import {Notification} from '../_redux/notification_center_types';
+
+export interface NotificationCenterSuccessfulResponse {
+  notifications: Notification[];
+  error?: undefined;
+}
+
+export interface NotificationCenterFailedResponse {
+  notifications?: Notification[];
+  error: string;
+}
+
+export type NotificationCenterResponse =
+  | NotificationCenterSuccessfulResponse
+  | NotificationCenterFailedResponse;
+
+export abstract class NotificationCenterDataSource {
+  abstract fetchNotification(): Observable<{
```

Comment:
nit: I'd tend to call this `fetchNotifications` since it typically returns more than one notification